File: src/main/java/net/igneo/icv/enchantmentActions/enchantManagers/weapon/BreakthroughManager.java
package net.igneo.icv.enchantmentActions.enchantManagers.weapon;

import net.igneo.icv.ICV;
import net.igneo.icv.enchantment.EnchantType;
import net.igneo.icv.entity.ICVEntity;
import net.igneo.icv.init.ICVUtils;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.player.Player;

public class BreakthroughManager extends WeaponEnchantManager {
    public BreakthroughManager(Player player) {
        super(EnchantType.WEAPON, player, new ResourceLocation(ICV.MOD_ID, "dual_handed_slash_cross"));
    }
    
    @Override
    public void activate() {
        super.activate();
        ICVEntity closestEntity = null;
        double lowestDistance = 99999;
        for (Entity e : ICVUtils.collectEntitiesBox(player.level(), player.position().add(player.getLookAngle().scale(0.3)), 2)) {
            if (e instanceof ICVEntity entity) {
                if (player.distanceTo(entity) < lowestDistance) {
                    closestEntity = entity;
                    lowestDistance = player.distanceTo(entity);
                }
            }
        }
        if (closestEntity != null) closestEntity.discard();
    }
}

File: src/main/java/net/igneo/icv/enchantmentActions/enchantManagers/weapon/BreakthroughManager.kt
package net.igneo.icv.enchantmentActions.enchantManagers.weapon

import net.igneo.icv.ICV
import net.igneo.icv.Utils.collectEntitiesBox
import net.igneo.icv.enchantment.EnchantType
import net.igneo.icv.entity.ICVEntity
import net.minecraft.resources.ResourceLocation
import net.minecraft.world.entity.player.Player

class BreakthroughManager(player: Player?) :
    WeaponEnchantManager(
        EnchantType.WEAPON,
        player!!,
        ResourceLocation(ICV.MOD_ID, "dual_handed_slash_cross")
    ) {
    override fun activate() {
        super.activate()
        var closestEntity: ICVEntity? = null
        var lowestDistance = 99999.0
        for (e in collectEntitiesBox(
            player.level(),
            player.position().add(player.lookAngle.scale(0.3)),
            2.0
        )) {
            if (e is ICVEntity) {
                if (player.distanceTo(e) < lowestDistance) {
                    closestEntity = e
                    lowestDistance = player.distanceTo(e).toDouble()
                }
            }
        }
        closestEntity?.discard()
    }
}
